src/components/ContactUsForm/ContactUs.jsx

```diff
@@ -0,0 +1,135 @@
+//       `,
+//       { businessId, message, timestamp, beenDismissed },
+//     );
+//     res.status(200).json({
+//       status: 'Success',
+//     });
+//   } catch (err) {
+//     res.status(500).send(err.message);
+//   }
+// });
+
+
+
+
+const ContactUs = () => {
+  const { backend } = useBackend();
+  const [ID, setID] = useState(0);
+  const [name, setName] = useState("business?"); // name of business
+  const [text, setText] = useState("");
+  const [checkedItems, setCheckedItems] = useState([false, false, false, false, false]);
+  const [formData, setFormData] = useState({ // idk whats in the form data
+    business_id: 0,
+    message: null,
+    timestamp: new Date().toLocaleString('en-US', { timeZone: 'America/Los_Angeles' }),
+    been_dismissed: false,
+  });
+
+
+  const handleCheck = (index, newValue) => {
+    setCheckedItems((checkedItems) => {
+      const newArray = [...checkedItems];
+      newArray[index] = newValue;
+      return newArray;
+    });
+
+
+  };
+
+   const SubmitForm = async event => {
+    event.preventDefault();
+    //console.log(formData);
+    const result = await backend.post('/notification/', formData);
+    console.log(result);
+   }
+
+   const updateForm = (inputValue) => { // how do you make this run first before submit?
+    const message = setMessage(inputValue);
+    setFormData({
+      business_id: 0,
+      message: message,
+      timestamp: new Date().toLocaleString('en-US', { timeZone: 'America/Los_Angeles' }), //idk
+      been_dismissed: false,
+    });
+   }
+
+   const handleInputChange = (e) => {
+    let inputValue = e.target.value;
+    updateForm(inputValue);
+    setText(inputValue);
+  }
+
+  const setMessage = (text) => {
+    const textCopy = text;
+    const checkedThingies = ['Stickers', 'Posters', 'Business Cards', 'Something', 'Joshua Lipton'];
+    const preMessage = `${name} is requesting: \n`; // business name... hard-coded right now
+    const updatedMessage = checkedThingies
+      .filter((_, index) => checkedItems[index])
+      .join('\n-');
+    return `${preMessage}-${updatedMessage}\nNotes:\n${textCopy}`;
+  };
+
+   return (
+    <>
+      <Flex>
+        <Box>
+            <Text>Supply Request</Text>
+            <Stack>
+                <Checkbox onChange={(e) => handleCheck(0, e.target.checked)}>Stickers</Checkbox>
+                <Checkbox onChange={(e) => handleCheck(1, e.target.checked)}>Posters</Checkbox>
+                <Checkbox onChange={(e) => handleCheck(2, e.target.checked)}>Business Cards</Checkbox>
+                <Checkbox onChange={(e) => handleCheck(3, e.target.checked)}>Something</Checkbox>
+                <Checkbox onChange={(e) => handleCheck(4, e.target.checked)}>Joshua Lipton</Checkbox>
+            </Stack>
+            <Text>Notes</Text>
+            <Textarea
+            value={text}
+            onChange={handleInputChange}
+            />
+            <Button onClick={SubmitForm}>Submit</Button>
```

Comment:
you can set up calling both updateFormData and SubmitForm in this onClick
```
() => {
    UpdateForm()
    SubmitForm()
}
```

Reply:
for UpdateForm we can scrap UpdateForm and construct the message at the end rather than every single time the form is updated